src/util/environment.ts

```diff
@@ -0,0 +1,332 @@
+import packageJson from "../../package.json";
+
+let os: any;
+try {
+  os = require("node:os");
+} catch (_) {
+  os = undefined;
+}
+
+/**
+ * Predicate to determine if we're executing in a Node environment or not
+ * @internal
+ */
+export const isNode = (): boolean => {
+  if (typeof process !== "undefined" && process.release?.name === "node") {
+    try {
+      require("node:http2");
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+  return false;
+};
```

Comment:
this does not tell us if its node - but if its node with HTTP2.

We should use slightly different algorithms for fingerprinting vs determining a default client.

This one can be used for the default client - and can stay where it was (perhaps with a different name like useNodeHttp2) and a new one can just determine if its node or not.